### PR TITLE
feat: hide all chats tab

### PR DESCRIPTION
**Pull request by Henvy-Mango**
# Tittle Here

Hide All Chats Tab
## Description

feature #77
## Check List

- [x] I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] My code follows the code style of this project
- [x] I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

**Comment by qwq233** [comment]
已经做了 没push 周末push

**Comment by qwq233** [comment]
算了 现在也行

File: TMessagesProj/src/main/java/org/telegram/ui/Components/FilterTabsView.java [hunk 1]
```diff
@@ -1778,15 +1802,15 @@ public boolean isLongPressDragEnabled() {
...
-            if (MessagesController.getInstance(UserConfig.selectedAccount).premiumLocked && (!isEditing || (viewHolder.getAdapterPosition() == 0 && tabs.get(0).isDefault && !UserConfig.getInstance(UserConfig.selectedAccount).isPremium()))) {
+            if (!ConfigManager.getBooleanOrFalse(Defines.hideAllTab) && (!isEditing || (viewHolder.getAdapterPosition() == 0 && tabs.get(0).isDefault && !UserConfig.getInstance(UserConfig.selectedAccount).isPremium()))) {
                 return makeMovementFlags(0, 0);
             }
             return makeMovementFlags(ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT, 0);
         }
 
         @Override
         public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder source, RecyclerView.ViewHolder target) {
-            if (MessagesController.getInstance(UserConfig.selectedAccount).premiumLocked && ((source.getAdapterPosition() == 0 || target.getAdapterPosition() == 0) && !UserConfig.getInstance(UserConfig.selectedAccount).isPremium())) {
```
Comment (qwq233) [inline TMessagesProj/src/main/java/org/telegram/ui/Components/FilterTabsView.java:1789]: `MessagesController.getInstance(UserConfig.selectedAccount).premiumLocked`
保留一下

File: TMessagesProj/src/main/java/org/telegram/ui/Components/FilterTabsView.java [hunk 2]
```diff
@@ -1778,15 +1802,15 @@ public boolean isLongPressDragEnabled() {
 
         @Override
         public int getMovementFlags(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder) {
-            if (MessagesController.getInstance(UserConfig.selectedAccount).premiumLocked && (!isEditing || (viewHolder.getAdapterPosition() == 0 && tabs.get(0).isDefault && !UserConfig.getInstance(UserConfig.selectedAccount).isPremium()))) {
```
Comment (qwq233) [inline TMessagesProj/src/main/java/org/telegram/ui/Components/FilterTabsView.java:1781]: 同下

**Review by qwq233 — CHANGES_REQUESTED** [review]

File: TMessagesProj/src/main/java/org/telegram/ui/Components/FilterTabsView.java (hunk 1 above)
Comment (Henvy-Mango) [reply TMessagesProj/src/main/java/org/telegram/ui/Components/FilterTabsView.java:1789]: ok

**Comment by Henvy-Mango** [comment]
可以合并了

**Review by qwq233 — APPROVED** [review]

**Pull request merged**